src/betterproto/__init__.py

```diff
@@ -343,20 +343,26 @@ def _pack_fmt(proto_type: str) -> str:
     }[proto_type]
 
 
-def encode_varint(value: int) -> bytes:
-    """Encodes a single varint value for serialization."""
-    b: List[int] = []
-
+def dump_varint(value: int, stream: BinaryIO) -> None:
+    """Encodes a single varint and dumps it into the provided stream."""
     if value < 0:
         value += 1 << 64
 
     bits = value & 0x7F
     value >>= 7
     while value:
-        b.append(0x80 | bits)
+        stream.write((0x80 | bits).to_bytes())
         bits = value & 0x7F
         value >>= 7
-    return bytes(b + [bits])
+    stream.write(bits.to_bytes())
+
+
+def encode_varint(value: int) -> bytes:
+    """Encodes a single varint value for serialization."""
+    with BytesIO() as stream:
+        dump_varint(value, stream)
+        bs = stream.getvalue()
+    return bs
```

Comment:
You can returned directly from the context manager:

```suggestion
        return stream.getvalue()
```

Reply:
Fixed